### PR TITLE
Update rubocop gem to 0.90.0

### LINKED ISSUE #8302: Update dependency constraint to allow for rubocop v0.88.0
Hey there! :wave:

I noticed that the constraint you have for rubocop doesn't allow for the latest version to be used.

The constraint I found was `~> 0.87.1`, and the latest version available is `0.88.0`.

Can you look into updating that constraint so our users can use the latest and greatest version? Thanks! :revolving_hearts:

### LINKED ISSUE #8327: Update dependency constraint to allow for rubocop v0.89.0
Hey there! :wave:

I noticed that the constraint you have for rubocop doesn't allow for the latest version to be used.

The constraint I found was `~> 0.87.1`, and the latest version available is `0.89.0`.

Can you look into updating that constraint so our users can use the latest and greatest version? Thanks! :revolving_hearts:

### LINKED ISSUE #8333: Update dependency constraint to allow for rubocop v0.89.1
Hey there! :wave:

I noticed that the constraint you have for rubocop doesn't allow for the latest version to be used.

The constraint I found was `~> 0.87.1`, and the latest version available is `0.89.1`.

Can you look into updating that constraint so our users can use the latest and greatest version? Thanks! :revolving_hearts:

### LINKED ISSUE #8366: Update dependency constraint to allow for rubocop v0.90.0
Hey there! :wave:

I noticed that the constraint you have for rubocop doesn't allow for the latest version to be used.

The constraint I found was `~> 0.87.1`, and the latest version available is `0.90.0`.

Can you look into updating that constraint so our users can use the latest and greatest version? Thanks! :revolving_hearts:

### PR DESCRIPTION
**Pull request by jnozsc**
This is a 🙋 feature or enhancement. 

## Summary

Update rubocop gem from version 0.87.1 to 0.90.0

## Context

Closes #8302
Closes #8327
Closes #8327
Closes #8333
Closes #8366

### CODE REVIEW

**Comment by jnozsc** [comment]
well, it seems these failures from `SUITE: test / OS: windows-latest (pull_request)` are related to symlink

**Comment by ashmaroli** [comment]
@jnozsc Hi!, Are you still interested in merging this contribution?
If yes, please update the branch to use RuboCop-0.90.x instead.
If you're not interested, let us know anyways.

**Comment by ashmaroli** [comment]
Thank you @jnozsc 
Some new offenses are being reported now:
https://travis-ci.org/github/jekyll/jekyll/jobs/723837262

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -14,3 +14,9 @@ Layout/SpaceAroundOperators:
   Exclude:
     - 'lib/jekyll/commands/build.rb'
     - 'test/test_configuration.rb'
+
+# Offense count: 1
+# Cop supports --auto-correct.
+Lint/RedundantCopDisableDirective:
+  Exclude:
+    - 'lib/jekyll/document.rb'
```
Comment (ashmaroli) [inline .rubocop_todo.yml:22]: If RuboCop thinks it is redundant, why are we not removing the directive?

Comment (jnozsc) [reply .rubocop_todo.yml]: fixed, thanks

**Review by DirtyF — APPROVED** [review]

**Comment by ashmaroli** [comment]
TODO before merging this:
- [ ] Enable and fix offenses reported by `rubocop-performance` cop (**via a separate pull request**)
- [ ] Configure cops added in RuboCop versions after v0.88 (**by appending to current pull request**)

**Review by DirtyF — APPROVED** [review]
🤖

**Comment by DirtyF** [comment]
@jekyll: merge +dev

**Pull request merged**